core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -497,6 +500,37 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  private def getExistingWorkspaceInvites(workspaceName: WorkspaceName) = {
+    dataSource.inTransaction { dataAccess =>
+      withWorkspaceContext(workspaceName, dataAccess) { workspaceContext =>
+        dataAccess.workspaceQuery.getInvites(workspaceContext.workspaceId).map { pairs =>
+          pairs.map(pair => WorkspaceACLUpdate(pair._1, pair._2))
+        }
+      }
+    }
+  }
+
+  private def deleteWorkspaceInvites(invites: Seq[WorkspaceACLUpdate], existingInvites: Seq[WorkspaceACLUpdate], workspaceName: WorkspaceName) = {
+    dataSource.inTransaction { dataAccess =>
+      withWorkspaceContext(workspaceName, dataAccess) { workspaceContext =>
+        val removals = invites.filter(_.accessLevel.equals(WorkspaceAccessLevels.NoAccess))
```

Comment:
IntelliJ is showing a warning for `_.accessLevel.equals(WorkspaceAccessLevels.NoAccess)`:

```
Comparing unrelated types less... (⌘F1) 
Detects comparisons (== and !=) of expressions which cannot be of the same type.
```
Manual inspection looks fine to me, but maybe there's something subtle going on here that I'm missing.

Interestingly, `_.accessLevel == WorkspaceAccessLevels.NoAccess` eliminates the warning. Coming from a Java background, this feels opposite to me, but I think (not 100% positive) that that equality expression will do the right thing in this case.